plutus-tx/src/PlutusTx/Blueprint/TH.hs

```diff
@@ -40,10 +44,90 @@ makeHasSchemaInstance dataTypeName indices = do
   let nonOverlapInstance = TH.InstanceD Nothing
 
   -- Lookup indices for all constructors of a data type.
-  indexedCons <- for (TH.datatypeCons dataTypeInfo) $ \ctorInfo ->
-    case lookup (TH.constructorName ctorInfo) indices of
-      Just index -> pure (ctorInfo, index)
-      Nothing    -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+  indexedCons :: [(TH.ConstructorInfo, SchemaInfo, Natural)] <- do
+    typeSchemaAnns <- lookupSchemaAnns dataTypeName
+    for (TH.datatypeCons dataTypeInfo) $ \ctorInfo -> do
+      let ctorName = TH.constructorName ctorInfo
+      case lookup ctorName indices of
+        Nothing -> fail $ "No index given for constructor " ++ show (TH.constructorName ctorInfo)
+        Just index -> do
+          {- Its possible to annotate a type as well as its constructors with schema annotations.
+             Consider the following example:
+
+            {-# ANN MkLeft (SchemaDescription "Constructor: MkLeft") #-}
+            {-# ANN MkRight (SchemaDescription "Constructor: MkRight") #-}
+            data Or = MkLeft | MkRight
+
+            With these annotations the following schema will be generated:
+
+            { ...
+              "Or": {
+                "oneOf": [
+                  {
+                    "description": "Constructor: MkLeft",
+                    "dataType": "constructor",
+                    "fields": [],
+                    "index": 0
+                  },
+                  {
+                    "description": "Constructor: MkRight",
+                    "dataType": "constructor",
+                    "fields": [],
+                    "index": 1
+                  }
+                ]
+              },
+            ...
+            }
+
+            Such schema lists all constructors of a type with their respective annotations,
+            however, there is no place to put annotations for the type itself.
+
+            For this reason, we merge annotations for the type and its constructor:
+
+            {-# ANN type Or (SchemaTitle "Type: Or") #-}
+            {-# ANN MkLeft (SchemaDescription "Constructor: MkLeft") #-}
+            {-# ANN MkRight (SchemaDescription "Constructor: MkRight") #-}
+            data Or = MkLeft | MkRight
+
+            becomes:
+
+            { ...
+              "Or": {
+                "oneOf": [
+                  {
+                    "title": "Type: Or",
+                    "description": "Constructor: MkLeft",
```

Comment:
I'm sorry, this still reads like nonsense to me. To me, "title/description" means "name/description". In what way is the "name" of the constructor "Or"? It just isn't! It's "MkLeft"!